engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/callcaching/package.scala

```diff
@@ -1,28 +1,71 @@
 package cromwell.engine.workflow.lifecycle.execution
 
+import cromwell.database.sql.MetaInfoId
+
 package object callcaching {
 
   // TODO: Find somewhere better for all these?
   case class HashKey(key: String)
   case class HashValue(value: String)
   case class HashResult(hashKey: HashKey, hashValue: HashValue)
 
-  private[callcaching] case class CacheResultMatchesForHashes(hashResults: Iterable[HashResult], cacheResultIds: Set[Int])
+  object HashValue {
+    implicit class StringMd5er(unhashedString: String) {
+      def md5HashValue: HashValue = {
+        val hashBytes = java.security.MessageDigest.getInstance("MD5").digest(unhashedString.getBytes)
```

Comment:
Looking at the docs, I think you need a new instance per hash (otherwise everything aggregates with the previous hash result)